ksql-engine/src/main/java/io/confluent/ksql/util/QueryIdGenerator.java

```diff
@@ -16,12 +16,19 @@
 
 package io.confluent.ksql.util;
 
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
+
 public class QueryIdGenerator {
 
-  private long queryIdCounter = -1;
+  private final AtomicLong queryIdCounter = new AtomicLong(0);
+  private final String postfix;
 
-  public long getNextId() {
-    return ++queryIdCounter;
+  public QueryIdGenerator(final String postfix) {
```

Comment:
Due to the validation following the same code path as the execution, we need to have separate instances of `QueryIdGenerator` for validation vs execution. Otherwise, the id's of queries will change with this PR.  To stop clashes in the query ids the two generate I've added a postfix.